src/main/java/emissary/place/VersionPlace.java

```diff
@@ -0,0 +1,76 @@
+package emissary.place;
+
+import emissary.core.IBaseDataObject;
+import emissary.core.ResourceException;
+import emissary.util.Version;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class VersionPlace extends ServiceProviderPlace {
+
+    private static final String EMISSARY_VERSION = "EMISSARY_VERSION";
+    private boolean includeDate;
+    private final Version version = new Version();
+
+    /**
+     * Create the place from the specified config file or resource
+     *
+     * @param configInfo the config file or resource to use
+     * @param dir the name of the controlling directory to register with
+     * @param placeLoc string name of this place
+     */
+    public VersionPlace(String configInfo, String dir, String placeLoc) throws IOException {
+        super(configInfo, dir, placeLoc);
+        configurePlace();
+    }
+
+    /**
+     * Create the place from the specified config stream data
+     *
+     * @param configInfo the config file or resource to use
+     * @param dir the name of the controlling directory to register with
+     * @param placeLoc string name of this place
+     */
+    public VersionPlace(InputStream configInfo, String dir, String placeLoc) throws IOException {
+        super(configInfo, dir, placeLoc);
+        configurePlace();
+    }
+
+    /**
+     * Create the place from the specified config stream data
+     *
+     * @param configInfo the config file or resource to use
+     */
+    public VersionPlace(InputStream configInfo) throws IOException {
+        super(configInfo);
+        configurePlace();
+    }
+
+    /**
+     * Create with the default configuration
+     */
+    public VersionPlace() throws IOException {
+        super();
+        configurePlace();
+    }
+
+    private void configurePlace() {
+        includeDate = configG.findBooleanEntry("INCLUDE_DATE", true);
+    }
+
+    @Override
+    public void process(IBaseDataObject payload) throws ResourceException {
+        payload.putParameter(EMISSARY_VERSION, getEmissaryVersion());
+    }
+
+    private String getEmissaryVersion() {
```

Comment:
Let's use this approach to populate a field in the `configurePlace` method, and then the `process` method can use that field value.

Reply:
yeah brain-farted there not to just doing that :laughing:, made the change